modules/tutorial-advanced-minimalist-fulcro/pages/index.adoc

```diff
@@ -0,0 +1,338 @@
+Inserting a stateful UI component between a parent-child entities::
+This section essentially explains why inserting a router component into your UI does not break the query. You should not need to do this yourself but it will help you understand routers.
+
+== Routing
+
+In a single page application, you want to enable the user to navigate to parts of the application and to only display these "active" parts. You might also want to only load the necessary data when you are are about to display the relevant part of the UI. Often you also want to bind the route to the URL. All this is the task of routing.
+
+Fulcro's link:{url-book}#_dynamic_router[Dynamic Routing] enables you to:
+
+1. Make Routers that have multiple target components and only display the active one (i.e. the one you have routed to)
+2. The routers can be nested, e.g. to display a particular "page" and then a particular "tab" within it
+3. Bind the route to the URL (Fulcro's routing is fully independent of the URL but the two can be hooked together)
+4. Load the data just before the component is about to be displayed
+5. The target component can dynamically decide whether to allow the user to route to it or to route out of it (e.g. you do not want to allow the user to leave a half-filled form until she either saves it or cancels the edit)
+6. Split the code so that parts of it will only be loaded when the part of the application they define is going to be displayed
+7. Only query for the props of its current target from the client DB, which is important for the performance of bigger apps
+8. Display a loading indicator after a short delay and to mark the route as failed if it does not complete within a time limit
+
+=== The bare essentials of Fulcro routing
+
+To add routing to your application, you need to:
+
+1. Create a router using `dr/defrouter` and give it the list of the target components, only one of which will be displayed at a time. Notice that there is no magic in `defrouter` - it just produces an old good `defsc`
+2. Give each target component a unique _route segment_ so that you can route to this target. You can think of the route segment as a relative URL. It starts with a non-empty literal string and may contain other strings and placeholders (keywords) for capturing dynamic _route parameters_ such as IDs. They do compose, if you have multiple levels of nested routers
+.. The component can use the `:will-enter` component option function to access the route parameters (which are always strings) and to translate them into the actual ident via `(dr/route-immediate <ident>)` or `route-deferred`
+3. Initialize everything and then route to the target you want displayed via `dr/change-route`
+
+Let's explore a minimalist example of routing. This is the UI that we are creating:
+
+image::bare-essentials-routing-ui-mockup.png["A mock up of a UI with a router (via wiredjs.com)"]
```

Comment:
Hukka: "link above the mockup image is broken"

JH: I know, it is OK. It will be fixed when we move from GH to the site proper.